### PR TITLE
feat: Add ``presence`` property to ``User``

### PR DESCRIPTION
**Pull request by Damego**
## About

This pull request adds `presence` property to `User`.
Can be ready to merge after merging #1004 

## Checklist

- [ ] I've ran `pre-commit` to format and lint the change(s) made.
- [x] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues) (If existent):.
  - resolves #
- I've made this pull request for/as: (check all that apply)
  - [x] Documentation
  - [ ] Breaking change
  - [x] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

File: interactions/api/models/user.py [hunk 1]
```diff
@@ -96,3 +99,15 @@ def banner_url(self) -> Optional[str]:
...
+    def presence(self) -> "Presence":
+        """
+        Returns the presence of the user.
+
+        :return: Presence of the user (None will be returned if not cached)
+        :rtype: Presence
+        """
+        from .gw import Presence
+
+        return self._client.cache[Presence].get(self.id)
```
Comment (mAxYoLo01) [inline interactions/api/models/user.py]: If `None` can be returned, make the return type and the typehint `Optional`?

Comment (Damego) [reply interactions/api/models/user.py]: Yeah you're right. I did add this

File: interactions/api/models/user.py [hunk 2]
```diff
@@ -96,3 +99,15 @@ def banner_url(self) -> Optional[str]:
...
         url += ".gif" if self.banner.startswith("a_") else ".png"
         return url
+
+    @property
+    def presence(self) -> Optional["Presence"]:
+        """
+        Returns the presence of the user.
+
+        :return: Presence of the user (None will be returned if not cached)
+        :rtype: Optional["Presence"]
```
Comment (mAxYoLo01) [inline interactions/api/models/user.py]:
```suggestion
        :rtype: Optional[Presence]
```
I think " are not needed in docstrings right?

**Comment by Damego** [comment]
oh forgot about this pr lol

**Review by sillyElvy — APPROVED** [review]

**Pull request merged**